### ISSUE TITLE
Add pre-commit hooks for linting

### ISSUE THREAD

**Issue by ShauryaAg** [adityabisoi/REST-fetch#69]
Add pre-commit hooks that runs linting before committing the file.

### Requirements:
 - [ ] Add `.pre-commit-config.yml` and define eslint and prettier hooks
 - [ ] Update the docs to install pre-commit hooks

### Reference: 
- https://pre-commit.com/

**Comment by Samridhi-98** [comment]
I would like to work on this issue, will you please assign this to me.

**Comment by ShauryaAg** [comment]
@Samridhi-98 Sure. You can work on this one, but please note this issue's PR will only be merged after #68 is done.

**Comment by Samridhi-98** [comment]
> @Samridhi-98 Sure. You can work on this one, but please note this issue's PR will only be merged after #68 is done.

Okay, no problem.

**Comment by rithgan** [comment]
Can i work on this if no one is assigned?

**Comment by ShauryaAg** [comment]
@rithgan Sure. You can take it. @Samridhi-98 is working on another issue.

**Comment by rithgan** [comment]
I have added the config file. There is a command on pre-commit which runs the prettier and eslint hooks on all the files. This will add the configuration as specified in prettier and eslint config files. Should I run this as well? This will cause indentation, whitespace like changes in all the files.

**Comment by ShauryaAg** [comment]
@rithgan Sure.
